tests/receivers/smartagent/prometheus-exporter/testdata/resource_metrics/httpd_negation_filtered.yaml

```diff
@@ -0,0 +1,7 @@
+resource_metrics:
+  - scope_metrics:
+      - metrics:
+          - attributes:
+              system.type: prometheus-exporter
+            name: istio_agent_go_gc_duration_seconds_count
```

Comment:
Unfortunately this won't detect the presence of additional undesired metrics given https://github.com/signalfx/splunk-otel-collector/blob/main/tests/testutils/telemetry/metrics.go#L310. Experimenting with restricting our tests w/ https://github.com/signalfx/splunk-otel-collector/pull/3265

Reply:
Hopefully the new restriction should be helpful since tests will now fail if unexpected metrics are received.

Reply:
Rebasing off latest main to pick up those changes, thanks :)